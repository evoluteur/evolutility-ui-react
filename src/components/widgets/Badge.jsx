import React from "react";
import PropTypes from "prop-types";

import "./Badge.scss";

const Badge = ({ text }) => (
  <div className="evo-badge">
    <span>{text}</span>
  </div>
);

export default Badge;

Badge.propTypes = { text: PropTypes.string.isRequired };
