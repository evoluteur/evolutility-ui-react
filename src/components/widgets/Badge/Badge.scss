@import "../../../variables";

.evo-badge {
  display: inline-block;
  height: 28px;
  min-width: 28px;
  padding: 5px 10px;
  border-radius: 14px;
  font-size: 14px;
  font-weight: 400;
  margin: 0 5px;
  text-align: center;
  border-style: solid;
  border-width: 1px;
}

.alert-default {
  color: black;
  background-color: $bgcolor-badge;
  border-color: silver;
}
