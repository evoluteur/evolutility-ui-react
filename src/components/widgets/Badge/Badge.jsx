import React from "react";
import PropTypes from "prop-types";

import "./Badge.scss";

const Badge = ({ text }) => (
  <div className="evo-badge" role="status" data-testid="badge">
    <span>{text}</span>
  </div>
);

export default Badge;

Badge.propTypes = {
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};
