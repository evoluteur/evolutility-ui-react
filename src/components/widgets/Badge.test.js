import { render, screen, fireEvent } from "@testing-library/react";
import Badge from "./Badge";
import renderer from 'react-test-renderer';

describe('Badge widget tests', () => {
    it('component renders correctly', () =>{
    const tree = renderer.create(<Badge text="Your Badge"/>).toJSON();
    expect(tree).toMatchSnapshot();
    });
    it('badge shows text string correctly',()=>{
    render(<Badge text="You got a badge!" />);
    const badge = screen.getByTestId("badge-test");
    expect(badge).toHaveTextContent("You got a badge!");
    });  
    it('badge shows number correctly',()=>{
    render(<Badge text={4152341234} />);
    const badge = screen.getByTestId("badge-test");
    expect(badge).toHaveTextContent(4152341234);
    }); 
    it('badge shows double byte text correctly',()=>{
    render(<Badge text="美味しい" />);
    const badge = screen.getByTestId("badge-test");
    expect(badge).toHaveTextContent("美味しい");
    });  
    it('badge shows special character text correctly',()=>{
    render(<Badge text="Ça va? #@$%" />);
    const badge = screen.getByTestId("badge-test");
    expect(badge).toHaveTextContent("Ça va? #@$%");
    }); 
}) 