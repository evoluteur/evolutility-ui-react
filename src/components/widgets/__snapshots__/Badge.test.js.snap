// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge widget tests component renders correctly 1`] = `
<div
  className="evo-badge"
  data-testid="badge-test"
  role="badge"
>
  <span>
    Your Badge
  </span>
</div>
`;
